### PR TITLE
Fix python prefix and shims search

### PR DESCRIPTION
**Pull request by AlongWY**
1. fix system python when pyenv NoVersion Error
2. fix linux arg0 make wrong python prefix

### CODE REVIEW

**Comment by kkysen** [comment]
Hi, thanks for the pull request!

What bug are you trying to fix?  Can you tell me what command and in what environment you're getting a wrong system python?  And what wrong prefix is the arg0 causing?

Also, it would be helpful if you could separate all the formatting changes into a separate commit so it's easier to see what the semantic changes are.

**Comment by AlongWY** [comment]
OS: Centos Linux 7 + Gentoo Preffix + Pyenv
1. When Pyenv::new() failed, the pyenv root maybe also exist, init by Self::System will cause exec pyenv shim python
2. I write a simple test, in source pyenv-python `python -c 'import sys; print(sys.prefix)'` it's wrong, it can't import packages, but the test version could. Then i delete the `cmd.arg0(arg0);`, it works!
```python
use anyhow::Context;
use apply::Apply;
use std::ffi::OsStr;
use std::io;
use std::os::unix::prelude::CommandExt;
use std::process::Command;

fn main() -> anyhow::Result<()> {
    let mut cmd = Command::new("~/.gentoo/usr/bin/python");
    cmd
        .exec()
        .apply(Err)
        .context("failed to run python subprocess")?;
    Ok(())
}
```

File: src/bin/python.rs [hunk 1]
```diff
@@ -185,9 +185,7 @@ impl Argv0Program {
     fn to_command(&self) -> Command {
         let mut args = env::args_os();
         let mut cmd = Command::new(self.argv0());
-        if let Some(arg0) = args.next() {
-            cmd.arg0(arg0);
```
Comment (AlongWY) [inline src/bin/python.rs:189]: The arg0 make `python -c 'import sys; print(sys.prefix)'` not equal `python --prefix`

File: src/lib.rs [hunk 2]
```diff
@@ -444,6 +444,30 @@ impl Python {
...
+                Err(system_python_error) => Err(PythonError {
+                    pyenv: PyenvError::NoExecutable {
+                        error,
+                        root,
+                        version,
+                        python_path,
+                    },
+                    system: system_python_error,
+                }),
+            },
```
Comment (AlongWY) [inline src/lib.rs:460]: When those PyenvError but Pyenv Exists

**Comment by kkysen** [comment]
Thanks for separating the formatting; it's a lot clearer now.

What are you trying to fix with `python --prefix` with arg0?  Arg0 doesn't affect the prefix at all which is based on the python path.  The arg0 setting code just invokes the python command with the appropriate name for arg0, but that shouldn't actually affect anything.

**Comment by AlongWY** [comment]
When i delete the arg0, the behavior is correct.
It also makes me confused, because your code works well in my macos, only wrong in my Centos Server

**Pull request closed without merging**